app/services/recommendation_service.py

```
@@ -27,23 +28,52 @@ def get_recommendations(
+            rec.get("paper_id"): rec.get("breakdown", {}) 
+            for rec in all_recommendations
+        }
+        
+        # 전체 후보군 총점 딕셔너리 (RL 학습용)
+        candidates_scores_dict = {
+            rec.get("paper_id"): rec.get("total_score", 0.0)
+            for rec in all_recommendations
+        }
+        
+        # 최종 추천된 6개 논문 ID 리스트 (RL Action)
+        final_display = [rec.get("paper_id") for rec in recommendations]
```

Ruff (0.14.6)

Undefined name `recommendations`

(F821)